### PR TITLE
bpf: host: optimize from-host's ICMPv6 path

### PR DESCRIPTION
**Pull request by julianwiedmann**
The ICMPv6 handling in handle_ipv6() is only required for the HostFW or by from-netdev. Exclude it otherwise.

This is a minor optimization for
dc9dfd72f2ae ("bpf: Re-introduce ICMPv6 NS responder on from-netdev").

### CODE REVIEW

File: bpf/bpf_host.c [hunk 1]
```diff
@@ -148,23 +148,26 @@ handle_ipv6(struct __ctx_buff *ctx, __u32 secctx __maybe_unused,
...
-		if (IS_ERR(ret))
-			return ret;
+		hdrlen = ipv6_hdrlen(ctx, &nexthdr);
+		if (hdrlen < 0)
+			return hdrlen;
+
+		if (likely(nexthdr == IPPROTO_ICMPV6)) {
+			ret = icmp6_host_handle(ctx, ETH_HLEN + hdrlen, ext_err, !from_host);
+			if (ret == SKIP_HOST_FIREWALL)
+				goto skip_host_firewall;
```
Comment (jschwinger233) [inline bpf/bpf_host.c:167]: nit: This `goto skip_host_firewall` can also go away with the `skip_host_firewall` label when `ENABLE_HOST_FIREWALL` is not defined. An optimization but not very valuable, feel free to resolve the conversation.

**Review by jschwinger233 — APPROVED** [review]
Thanks!

File: bpf/bpf_host.c (hunk 1 above)
Comment (julianwiedmann) [reply bpf/bpf_host.c:167]: ack, I *hope* the compiler is even able to prove that and apply dead-code elimination.

**Pull request merged**